FirebaseMessaging/Sources/FIRMessaging.m

```diff
@@ -617,6 +626,38 @@ - (void)deleteFCMTokenForSenderID:(nonnull NSString *)senderID
                                            handler:completion];
 }
 
+- (void)deleteWithCompletion:(void (^)(NSError *_Nullable))completion {
+  FIRMessaging_WEAKIFY(self);
+  [self.instanceID
```

Comment:
Optional nit: I always want to suggest using [Promises](https://github.com/google/promises) when I see several nested async operations.

Reply:
This is still temporary as Messaging underline just calling InstanceID calls. I think we can start applying Promises once we finish the large refactor of removing IID completely.